pkg/controllers/manifestwork.go

```diff
@@ -277,21 +277,26 @@ func (r *HypershiftDeploymentReconciler) deleteManifestworkWaitCleanUp(ctx conte
 
 func (r *HypershiftDeploymentReconciler) appendHostedClusterReferenceSecrets(ctx context.Context, providerSecret *corev1.Secret) loadManifest {
 	return func(hyd *hypdeployment.HypershiftDeployment, payload *[]workv1.Manifest) error {
-		var pullCreds *corev1.Secret
 		var err error
-		if !hyd.Spec.Infrastructure.Configure {
-			pullCreds, err = r.generateSecret(ctx,
-				types.NamespacedName{Name: hyd.Spec.HostedClusterSpec.PullSecret.Name,
-					Namespace: hyd.GetNamespace()})
-
-			if err != nil {
-				return fmt.Errorf("failed to duplicateSecret, err %w", err)
+		refSecrets := []*corev1.Secret{}
+
+		if len(hyd.Spec.HostedClusterSpec.PullSecret.Name) != 0 {
+			var pullCreds *corev1.Secret
+			if !hyd.Spec.Infrastructure.Configure {
+				pullCreds, err = r.generateSecret(ctx,
+					types.NamespacedName{Name: hyd.Spec.HostedClusterSpec.PullSecret.Name,
+						Namespace: hyd.GetNamespace()})
+
+				if err != nil {
+					return fmt.Errorf("failed to duplicate pull secret, err %w", err)
```

Comment:
Same comment as below. Log an error about what we're doing, but return the actual error for debugging.